TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/Tensorflow.java

```diff
@@ -1,49 +1,139 @@
+    private TFObjectDetector tfod; //declaring objectDetector
 
-    public Tensorflow(WebcamName name, int tfodMonitorId) {
-        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters();
+    public Tensorflow(WebcamName name, int tfodMonitorId) { //name tfod id
+        VuforiaLocalizer.Parameters parameters = new VuforiaLocalizer.Parameters(); //creating parameters
 
-        parameters.vuforiaLicenseKey = VUFORIA_KEY;
+        parameters.vuforiaLicenseKey = VUFORIA_KEY; //setting key
 
-        parameters.cameraName = name;
+        parameters.cameraName = name; //setting name to name
 
-        vuforia = ClassFactory.getInstance().createVuforia(parameters);
+        vuforia = ClassFactory.getInstance().createVuforia(parameters); //creating vuforia obj w/parameters
 
-        TFObjectDetector.Parameters tfodParameters = new TFObjectDetector.Parameters(tfodMonitorId);
-        tfodParameters.minResultConfidence = 0.3f;
-        tfod = ClassFactory.getInstance().createTFObjectDetector(tfodParameters, vuforia);
-        tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABEL_FIRST_ELEMENT, LABEL_SECOND_ELEMENT);
-        tfod.activate();
+        TFObjectDetector.Parameters tfodParameters = new TFObjectDetector.Parameters(tfodMonitorId); //creating parameters
+        tfodParameters.minResultConfidence = 0.3f; //minimumConfidenceNecessaryForActingOnDetection
+        tfod = ClassFactory.getInstance().createTFObjectDetector(tfodParameters, vuforia); //create objectDetector
+        tfod.loadModelFromAsset(TFOD_MODEL_ASSET, LABEL_FIRST_ELEMENT, LABEL_SECOND_ELEMENT); //loading models
+        tfod.activate(); //turnOn
+        //Hi, this is Anish, and I'm teaching Vincent and Ethan, how git works.
     }
 
-    public List<Recognition> getInference() {
+    public List<Recognition> getInference() { //get "image" back, a bunch of Recognitions - check out instance variables
         if (tfod != null) {
-            return tfod.getUpdatedRecognitions();
+            return tfod.getUpdatedRecognitions(); //Returns the list of recognitions, but only if they are different than the last call to {@link #getUpdatedRecognitions()}.
         }
         return null;
     }
 
     public void activate() {
-        tfod.activate();
-    }
+        if (tfod != null){
+            tfod.activate();
+        }
+
+    } //activate
 
     public void shutdown() {
-        tfod.shutdown();
+        if (tfod != null){
+            tfod.shutdown();
+        }
+    } //deactivate
+
+    public SquareState getTargetRegion(){
+        if (tfod == null){
+            return null;
+        }
+        List<Recognition> toOperateOffOf = getInference();
+        if (toOperateOffOf.size() == 1){
+            int i = 0;
+            for (Recognition recognitions : toOperateOffOf){
+                telemetry.addData(String.format("label (%d)", i), recognitions.getLabel());
```

Comment:
Telemetry code that you've been using for debugging shouldn't get merged into master. Telemetry really should be limited to passing useful or necessary information back to the team during competition. Better to Log.d whatever you're tracking, or just remove it once it's solid.